### PR TITLE
fix(dialog): adding subtitle

### PR DESCRIPTION
**Pull request by rachelbt**

### CODE REVIEW

File: libs/components/src/lib/dialog/README.md [hunk 1]
```diff
@@ -201,7 +202,7 @@ Use the footer `slot` in order to add action buttons to the bottom of the dialog
 </style>
 <vwc-dialog open
   headline="Dialog with footer"
-  text="To quote Michael Carini, although we should never apologize for being ourselves, we should apologize for the times that we are not.">
+  subtitle="this is an example of the dialog with stooted buttons inside footer">
```
Comment (olaf-k) [inline libs/components/src/lib/dialog/README.md:205]: I'm guessing `slotted`?

Comment (rachelbt) [reply libs/components/src/lib/dialog/README.md:205]: you caught me :) :)

File: libs/components/src/lib/dialog/dialog.template.ts [hunk 2]
```diff
@@ -102,17 +103,18 @@ export const DialogTemplate: (
 								<slot name="graphic">
 									${when(x => x.icon, icon(iconTag))}
 								</slot>
-								${when(x => x.headline, headline())}
+								<div class="headline-text">
```
Comment (olaf-k) [inline libs/components/src/lib/dialog/dialog.template.ts:106]: maybe wrap the `div` in a `when(x.headline || x.subtitle)` to skip its rendering when none of them are set

File: libs/components/src/lib/dialog/ui.test.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import type {Dialog} from './dialog';
 
 const components = ['dialog'];
 
-test('should show the component', async ({ page }: { page: Page }) => {
+test.only('should show the component', async ({ page }: { page: Page }) => {
```
[the remark this answers is not in the document]
Comment (rachelbt) [reply libs/components/src/lib/dialog/ui.test.ts:11]: :scream: shame on me!!

File: libs/components/src/lib/dialog/dialog.scss [hunk 4]
```diff
@@ -62,23 +62,32 @@
...
 
+
 			.icon {
 				font-size: #{size.$vvd-size-condensed};
 			}
 		}
 
 		.base.icon-placement-side & {
-			column-gap: calc(#{size.$vvd-size-condensed} - #{size.$vvd-size-super-condensed});
+			column-gap: calc(#{size.$vvd-size-condensed} / 2);
```
Comment (yinonov) [inline libs/components/src/lib/dialog/dialog.scss:72]: doesn't scale
in default density it changes the value from 8 to 16
in `-1` density it changes 4 to 14

that's a wild card

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/dialog/dialog.scss [hunk 5]
```diff
@@ -62,23 +62,32 @@
...
 				font-size: #{size.$vvd-size-extra-condensed};
 			}
 		}
 	}
+
+	&-text {
+		display: flex;
+		flex-direction: column;
+		column-gap: 4px;
+	}
```
Comment (yinonov) [inline libs/components/src/lib/dialog/dialog.scss:84]: what's that?

File: libs/components/src/lib/dialog/dialog.scss [hunk 6]
```diff
@@ -34,7 +34,7 @@
 	display: flex;
 	flex-direction: column;
 	padding: 24px;
-	gap: 8px;
+	gap: 12px;
```
Comment (yinonov) [inline libs/components/src/lib/dialog/dialog.scss:37]: this raises a question whether it was authored by design. doesn't relate to this scope. can we extract?

File: libs/components/src/lib/dialog/dialog.scss (hunk 4 above)
Comment (rachelbt) [reply libs/components/src/lib/dialog/dialog.scss:72]: so no dividing but subtraction yes?

File: libs/components/src/lib/dialog/dialog.template.ts [hunk 7]
```diff
@@ -102,17 +115,15 @@ export const DialogTemplate: (
 								<slot name="graphic">
 									${when(x => x.icon, icon(iconTag))}
 								</slot>
-								${when(x => x.headline, headline())}
+								${when(x => x.headline || x.subtitle, headlineText())}
```
Comment (yinonov) [inline libs/components/src/lib/dialog/dialog.template.ts:118]: why not make them independent from 1 another? can't this be resolved with easily via css?

File: libs/components/src/lib/dialog/dialog.scss (hunk 5 above)
Comment (rachelbt) [reply libs/components/src/lib/dialog/dialog.scss:84]: style for the headline text wrapper that contains the headline + subtitle

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/dialog/dialog.template.ts (hunk 7 above)
Comment (rachelbt) [reply libs/components/src/lib/dialog/dialog.template.ts:118]: they are independent - but if none of them is inserted by the consumer - no wrapper div is generated

File: libs/components/src/lib/dialog/dialog.scss (hunk 6 above)
Comment (rachelbt) [reply libs/components/src/lib/dialog/dialog.scss:37]: this is a fix - originally didn't give the right gap....

File: libs/components/src/lib/dialog/dialog.template.ts (hunk 7 above)
Comment (yinonov) [reply libs/components/src/lib/dialog/dialog.template.ts:118]: how complicated would it be to concede the wrapper in favor of a grid?

Comment (rachelbt) [reply libs/components/src/lib/dialog/dialog.template.ts:118]: I considered it but there are too many options that flex is a much better and fluid one, even though we are "paying" with another wrapper

File: libs/components/src/lib/dialog/dialog.scss (hunk 4 above)
Comment (yinonov) [reply libs/components/src/lib/dialog/dialog.scss:72]: subtraction scales with design unit. gap should still be researched later on for whether it should be fluid. but this is yet to be standardized

Comment (rachelbt) [reply libs/components/src/lib/dialog/dialog.scss:72]: so should I use definite unit (8px) or leave it as it is?

File: libs/components/src/lib/dialog/dialog.template.ts (hunk 7 above)
Comment (rachelbt) [reply libs/components/src/lib/dialog/dialog.template.ts]: @yinonov  - I changed the html structure of header, remover 2 wrappers, used `grid`

**Review by yinonov — APPROVED** [review]

**Pull request merged**